### PR TITLE
Add testing, and download from RUM unreleased modules

### PR DESCRIPTION
**Pull request by pablomartinezbernardo**
* Ability to `--skipDownload` in order to run with local installer binary
* Download module from manual aws releases that support RUM
* Small param unit tests
* Snapshot testing: Given some nginx configs, store snapshots to ensure the results are as expected and don't regress. Please review the snapshots as well.
* Integration testing. Boot up docker compose with some unix distros, run the installer and check that it succeeds. Run for alpine, AL2023 and bookworm
* Separate workflow for installer. GH diff is terrible for files that moved. The changes are:
  1. Move config.yml to continue_config.yml
  2. Create a new config.yml that always runs and populates `run-installer-workflow` depending on the files modified. [Reasoning explained here](https://circleci.com/docs/using-dynamic-configuration/#execute-specific-workflows-or-steps-based-on-which-files-are-modified).
  3. Add new `build-and-test-installer` workflow that only runs when installer files were modified
  4. All other workflows remain as before

### CODE REVIEW

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+     - installer_test:
+        matrix:
+          parameters:
+            base-dockerfile:
+            - 'bookworm.Dockerfile'
+            - 'alpine.Dockerfile'
+            - 'al2023.Dockerfile'
+        name: test installer with << matrix.base-dockerfile >> base image
+        requires:
+        - build installer on amd64
```
Comment (dmehala) [inline .circleci/continue_config.yml:778]: build-and-test-all is generated thanks to `bin/generate_job.rb`. Usually one appends the output of this script to this file. To maintain this, would it be possible to move this workflow before build-and-test-all?

Comment (dmehala) [inline .circleci/continue_config.yml:778]:
```suggestion
        requires:
        - build installer on amd64
        - build installer on arm64
```

**Review by dmehala — COMMENTED** [review]
Few comments there and there. I noticed the shell script is not tested, may I ask what is preventing us from testing it?

**Review by dmehala — APPROVED** [review]

File: installer/configurator/configurator.go [hunk 2]
```diff
@@ -54,9 +54,14 @@ func main() {
 	skipVerify := flag.Bool("skipVerify", false, "Skip verifying downloads")
 	verbose := flag.Bool("verbose", false, "Verbose output")
 	dryRun := flag.Bool("dryRun", false, "Dry run (no changes made)")
+	skipDownload := flag.Bool("skipDownload", false, "Skip the download of this installer and use a local binary instead")
```
[the remark this answers is not in the document]
Comment (pablomartinezbernardo) [reply installer/configurator/configurator.go:57]: > is this flag useful from the user POV?

Maybe? I can see a case where we provide a manually built binary for a customer. But that is not the actual reason: Flag parsing fails for unknown flags. As we are forwarding all arguments from the script, the go binary needs to know about `skipDownload` as well. We could use [ContinueOnError](https://pkg.go.dev/flag#ErrorHandling), but that would skip all unknown arguments, and I think it's worth failing on incorrect commands. We could also implement manual parsing, or [use an alternative flag package that supports hidden arguments](https://pkg.go.dev/github.com/jessevdk/go-flags#section-readme).

All that to say: this solves the script passing the parameter to the binary, plus I didn't see it as that much of a problem from the customer's POV

File: installer/configurator/configurator.go [hunk 3]
```diff
@@ -93,6 +98,6 @@ func main() {
 			handleError(err)
 		}
 
-		log.Info("Datadog NGINX module has been successfully installed and configured. Please restart NGINX for the changes to take effect")
+		log.Info("Datadog NGINX module has been successfully installed and configured. Please reload NGINX or restart the service for the changes to take effect")
```
[the remark this answers is not in the document]
Comment (pablomartinezbernardo) [reply installer/configurator/configurator.go:101]: Reload is enough to load the module, but restarting is obviously also an option

**Comment by pablomartinezbernardo** [comment]
> Few comments there and there. I noticed the shell script is not tested, may I ask what is preventing us from testing it?

Almost everything in the script is environment-dependent (Checking if a binary is installed, running uname, verifying the agent is running, downloading the installer and decompressing it...), which makes it a bad candidate for unit tests. Instead, it is mostly tested in the integration tests, for which the entrypoint is the script.

The only thing missing being tested there is the download of the installer, skipDownload is used because we want to use the installer that was just built.

And I do believe we should be testing that, so I added one integration test to test for the download: it runs the script without skipDownload and doesn't care about the installation result, only that the module was downloaded correctly.

**Pull request merged**